sound/soc/codecs/es8316.c

```diff
@@ -717,6 +717,38 @@ static int es8316_set_jack(struct snd_soc_component *component,
 	return 0;
 }
 
+static int es8316_resume(struct snd_soc_component *component){
+        /* The dapm fails and needs a hardcode write */
+        printk("es8316 resume");
+	snd_soc_component_write(component, 0x00, 0x80);
+	snd_soc_component_write(component, 0x01, 0x7f);
+	snd_soc_component_write(component, 0x0a, 0x00);
+	snd_soc_component_write(component, 0x0d, 0x00);
+	snd_soc_component_write(component, 0x14, 0x88);
+	snd_soc_component_write(component, 0x15, 0x00);
+	snd_soc_component_write(component, 0x17, 0x66);
+	snd_soc_component_write(component, 0x18, 0x00);
+	snd_soc_component_write(component, 0x19, 0x02);
+	snd_soc_component_write(component, 0x1a, 0x02);
+	snd_soc_component_write(component, 0x1c, 0x00);
+        snd_soc_component_write(component, 0x22, 0x00);
+        snd_soc_component_write(component, 0x2e, 0x00);
+        snd_soc_component_write(component, 0x2f, 0x00);
+        snd_soc_component_write(component, 0x30, 0x00);
+        snd_soc_component_write(component, 0x31, 0x00);
+        snd_soc_component_write(component, 0x32, 0x00);
+	snd_soc_component_write(component, 0x33, 0x00);
+	snd_soc_component_write(component, 0x34, 0x00);
+	snd_soc_component_write(component, 0x03, 0x32);
+	snd_soc_component_write(component, 0x0c, 0xff);
+	snd_soc_component_write(component, 0x16, 0xbb);
+	snd_soc_component_write(component, 0x23, 0xa0);
+	snd_soc_component_write(component, 0x27, 0x00);
+	snd_soc_component_write(component, 0x4e, 0x02);
```

Comment:
@yangxiaohua2009 In theory the codec driver .resume should do a regmap_sync() to force the registers to be restored back to the values set before suspend. 

@bardliao can you chime in here?

Reply:
@yangxiaohua2009 You can check the existing examples on how codec use regmap_sync() in the resume call back, and see if this style works for you. it is trivial to restore reg one by one.
- rt1011: https://github.com/thesofproject/linux/blob/d8dba897d90355cc598cfb87018634db6cf7d00d/sound/soc/codecs/rt1011.c#L2099